packages/wallet-adapter-core/src/WalletCore.ts

```diff
@@ -266,86 +281,141 @@ export class WalletCore extends EventEmitter<WalletCoreEvents> {
     } catch (error: any) {
       const errMsg =
         typeof error == "object" && "message" in error ? error.message : error;
       throw new WalletSignAndSubmitMessageError(errMsg).message;
     }
   }
 
-  /**
-   Sign transaction (doesnt submit to chain).
-   @param transaction
-   @param options max_gas_amount and gas_unit_limit
-   @return response from the wallet's signTransaction function
-   @throws WalletSignTransactionError
-   */
+  // AnyRawTransaction - V2 input
+  // Types.TransactionPayload - V1 input
   async signTransaction(
-    transaction: Types.TransactionPayload,
-    options?: TransactionOptions
-  ): Promise<Uint8Array | null> {
-    if (this._wallet && !("signTransaction" in this._wallet)) {
-      throw new WalletNotSupportedMethod(
-        `Sign Transaction is not supported by ${this.wallet?.name}`
-      ).message;
-    }
-
+    transactionOrPayload: AnyRawTransaction | Types.TransactionPayload,
+    options?: InputGenerateTransactionOptions
+  ): Promise<AccountAuthenticator> {
     try {
       this.doesWalletExist();
-      const response = await (this._wallet as any).signTransaction(
-        transaction,
-        options
+      // if input is AnyRawTransaction, i.e V2
+      if ("rawTransaction" in transactionOrPayload) {
+        if (this._wallet?.version !== "v2") {
+          throw new WalletNotSupportedMethod(
+            `Sign Transaction V2 is not supported by ${this.wallet?.name}`
+          ).message;
+        }
+        const accountAuthenticator = (this._wallet as any).signTransaction(
+          transactionOrPayload
+        );
+
+        return accountAuthenticator;
+      }
+
+      // check current signTransaction function exists
+      if (this._wallet && !("signTransaction" in this._wallet)) {
+        throw new WalletNotSupportedMethod(
+          `Sign Transaction is not supported by ${this.wallet?.name}`
+        ).message;
+      }
+
+      const response = await this.waletCoreV1.signTransaction(
+        transactionOrPayload as Types.TransactionPayload,
+        this._wallet,
+        {
+          max_gas_amount: options?.maxGasAmount
+            ? BigInt(options?.maxGasAmount)
+            : undefined,
+          gas_unit_price: options?.gasUnitPrice
+            ? BigInt(options?.gasUnitPrice)
+            : undefined,
+        }
       );
-      return response;
+      if (!response) {
+        throw new Error("error");
+      }
+
+      // Convert retuned bcs serialized SignedTransaction into V2 AccountAuthenticator
+      const deserializer1 = new BCS.Deserializer(response);
+      const deserializedSignature =
+        TxnBuilderTypes.SignedTransaction.deserialize(deserializer1);
+      const transactionAuthenticator =
+        deserializedSignature.authenticator as TxnBuilderTypes.TransactionAuthenticatorEd25519;
+
+      const publicKey = transactionAuthenticator.public_key.value;
+      const signature = transactionAuthenticator.signature.value;
```

Comment:
this is where it is a bit weird, when trying to use this publicKey, signature and message to verifyMessage - verification fails. Failure is happening in both SDK V1 and V2.
```
const ed25519 = new Ed25519PublicKey(publicKey);
const verify = ed25519.verifyMessage(response, signature);
```

Consider the `signTransaction` function is being used with wallets we can assume it works? so 2 failure options
1. verifyMessage is wrong
2. something in the deserialization process is wrong

cc @xbtmatt @hardsetting

Reply:
We can look together at the derserialized bytes to see what's going on here, but it should be pretty simple to separate and determine.

The other possibility is if this is a signed message and not a transaction, is it might not have all of the data for the message to verify.